src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -27,44 +27,33 @@
 @WebServlet("/user")
 public class UserDataServlet extends HttpServlet {
 
-  /** Responsible for creating new User. */
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /** Responsible for add a new User into Datastore */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     // Get the input from the form.
-    String name = getParameter(request, "name", "");
-    String email = getParameter(request, "email", "");
-    String university = getParameter(request, "university", "");
+    String name = request.getParameter("name");
+    String email = request.getParameter("email");
+    String university = request.getParameter("university");
     String subscription = request.getParameter("email-notif");
-    boolean subscribe;
+    boolean subscribe = true;
 
-    if (subscription.equals("subscribe")) {
-      subscribe = true;
-    } else if (subscription.equals("unsubscribe")) {
+    // Check if user wants to unsubscribe.
+    if (subscription.equals("unsubscribe")) {
       subscribe = false;
     }
 
     // Create an Entity type User.
-    Entity userEntity = new Entity("User");
+    Entity userEntity = new Entity("User", email);
```

Comment:
Nit. In general, omit comments that are clear from the code. Usually, comments should focus on adding clarity to code e.g. explaining why some code is written a particular way.

Reply:
Gotcha!